File: archive/k/kotlin/LongestWord.kt
fun main(args: Array<String>) {

    fun longestWord(sentence: String): Any {
        // if sentence is empty, ask for a String
        if (sentence.length == 0){
            return "Usage: please provide a string"
        } else {
            // split sentence from these delimeters and put resulting strings into words list
            var words = sentence.split(" ", "\t", "\n", "\r")
            var longest = 0
            // iterate through words list and compare each word length to longest var
            // if word length is larger, then the var longest will be assigned the word length
            for (word in words){
                when {
                    word.length > longest -> longest = word.length
                }
            }
            // return var longest which holds the largest string length in the sentence parameter
            return longest
        }
    }    

    // if console input is null, ask for String
    if (args.isNullOrEmpty()){
        println("Usage: please provide a string")
    } else {
        // if console input is not null, then find longestWord of input String
        println(longestWord(args[0]))    
    }
    
}

File: archive/s/swift/longest-word.swift
import Foundation

if (CommandLine.arguments.count < 2) {
    print("Usage: please provide a string")
}
else 
{
    var sentence = CommandLine.arguments[1]   
    sentence = sentence.replacingOccurrences(of: "\n", with: "")    //removing the break line if it contains any
    longestWord(input : sentence)
}

func longestWord(input : String) -> Void
{
    var longest = 0
    var testWord = ""

    if(input == "")
    {
         print("Usage: please provide a string")    //checking for empty string
    }

    else 
    {
        var substrings: [Substring] = []    //array to hold the array of the input strings
        substrings = input.split(separator: " ")        //splitting the array by spaces
        
        for word in substrings    //iterate through the array
        {
            testWord = word.trimmingCharacters(in: CharacterSet(charactersIn: " "))
            if(testWord.count > longest)    
            {
                longest = testWord.count    //obtaining the longest count of words
            }
        }
        print(longest)
    }
}
